File: chapters/fundamental_algorithms/euclidean_algorithm/code/python2/euclidean_example.py
def euclid_mod(a, b):

    a = abs(a)
    b = abs(b)
    temp = 0

    while b > 0:
        temp = b
        b = a % b
        a = temp

    return a

def euclid_sub(a, b):

    a = abs(a)
    b = abs(b)

    while a != b:
        if a > b:
            a = a - b
        else:
            b = b - a

    return a

print euclid_mod(64 * 67, 64 * 81)
print euclid_sub(128 * 12, 128 * 77)

File: chapters/fundamental_algorithms/euclidean_algorithm/code/javascript/euclidean_example.js
<!DOCTYPE html>
<html>
<body>
<script>
function euclid_mod(a, b){
    a = Math.abs(a);
    b = Math.abs(b);

    var temp;
    while (b != 0){
        temp = b;
        b = a%b;
        a = temp;
    }

    return a;
}

function euclid_sub(a, b){
    a = Math.abs(a);
    b = Math.abs(b);

    while (a != b){
        if (a > b){
            a = a - b;
        }
        else{
            b = b - a;
        }
    }

    return a;
}

document.write(euclid_mod(64*67, 64*81) + "<br>");
document.write(euclid_sub(128*12, 128*77) + "<br>");
</script>
</body>
</html>

